crates/bevy_gizmos/src/gizmos.rs

```diff
@@ -1,57 +1,125 @@
 //! A module for the [`Gizmos`] [`SystemParam`].
 
-use std::{f32::consts::TAU, iter};
+use std::{f32::consts::TAU, iter, marker::PhantomData};
 
 use bevy_ecs::{
-    system::{Deferred, Resource, SystemBuffer, SystemMeta, SystemParam},
-    world::World,
+    component::Tick,
+    system::{Deferred, ReadOnlySystemParam, Res, Resource, SystemBuffer, SystemMeta, SystemParam},
+    world::{unsafe_world_cell::UnsafeWorldCell, World},
 };
 use bevy_math::{Mat2, Quat, Vec2, Vec3};
 use bevy_render::color::Color;
 use bevy_transform::TransformPoint;
 
+use crate::{
+    config::GizmoConfigGroup,
+    config::{DefaultGizmoGroup, GizmoConfigStore},
+    prelude::GizmoConfig,
+};
+
 type PositionItem = [f32; 3];
 type ColorItem = [f32; 4];
 
 const DEFAULT_CIRCLE_SEGMENTS: usize = 32;
 
 #[derive(Resource, Default)]
-pub(crate) struct GizmoStorage {
+pub(crate) struct GizmoStorage<T: GizmoConfigGroup> {
     pub list_positions: Vec<PositionItem>,
     pub list_colors: Vec<ColorItem>,
     pub strip_positions: Vec<PositionItem>,
     pub strip_colors: Vec<ColorItem>,
+    marker: PhantomData<T>,
 }
 
 /// A [`SystemParam`] for drawing gizmos.
 ///
 /// They are drawn in immediate mode, which means they will be rendered only for
 /// the frames in which they are spawned.
 /// Gizmos should be spawned before the [`Last`](bevy_app::Last) schedule to ensure they are drawn.
-#[derive(SystemParam)]
-pub struct Gizmos<'s> {
-    buffer: Deferred<'s, GizmoBuffer>,
+pub struct Gizmos<'w, 's, T: GizmoConfigGroup = DefaultGizmoGroup> {
+    buffer: Deferred<'s, GizmoBuffer<T>>,
+    pub(crate) enabled: bool,
+    /// The currently used [`GizmoConfig`]
+    pub config: &'w GizmoConfig,
+    /// The currently used [`GizmoConfigGroup`]
+    pub config_ext: &'w T,
 }
 
+const _: () = {
```

Comment:
Can remove the `const _: () = { … }` as it only exists to avoid exposing the types when using the derive macro. Here, they are already written out, and not exported, so it's fine.